indsl/data_quality/gaps_classification.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023 Cognite AS
+from typing import List
+
+import numpy as np
+import pandas as pd
+
+from indsl.exceptions import SCIKIT_LEARN_REQUIRED
+from indsl.type_check import check_types
+from indsl.validations import validate_series_has_time_index, validate_series_is_not_empty
+
+
+@check_types
+def gaps_classification(x: pd.Series, eps: float = 0.5, min_samples: int = 2, std_thresholds: List[int] = [1, 2, 3]):
+    """Gaps Classification.
+
+    Classify gaps in a time series dataset into categories based on duration and statistical properties.
```

Comment:
This description is going to be visible for charts end-user if you are planning to expose this function. Thus, make sure it is clear and user friendly